Backpack-SwiftUI/Calendar/Classes/Range/Cells/RangeSelectionCalendarDayCell.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+struct RangeSelectionCalendarDayCell: View {
+    let date: Date
+    @Binding var selection: ClosedRange<Date>?
+    let calendar: Calendar
+    
+    var body: some View {
+        if let selection, selection.contains(date) {
+            if selection.lowerBound == selection.upperBound {
+                LowerAndUpperBoundSelectedCell(calendar: calendar, date: date)
+            } else if date == selection.lowerBound {
+                LowerBoundSelectedCell(calendar: calendar, date: date)
+            } else if date == selection.upperBound {
+                UpperBoundSelectedCell(calendar: calendar, date: date)
+            } else {
+                InbetweenSelectionCell(calendar: calendar, date: date)
```

Comment:
I'm trying to consider the pros / cons of making individual cells like this versus have a view modifier to apply these various states to a common calendar cell.
I guess the way you have it here with distinct View structs gives some simplicity, and clarity when reading this, but it implies more duplication, eg we have to create a `BPKText()` view in each cell, and apply mostly the same styling to it each time. 
Just wondering if you considered making view modifiers to represent each of these cell types, which could be applied to a common view, or if you think that would end up being too painful / not working?

Reply:
I considered it yes. I ended up with different views for each day cell type to make them not depend on each other too much and have more flexibility when adding more cell types (for example, the component needs to support (not in this PR) adding prices to the day cells)

So flexibility here is more worth having than some duplication.

Duplication is not really happening much though here (except for maybe the fact it's using a BPKText with with lineLimit=1) and a modifier would be slightly harder to use for this usecase, in general, conditionally adding modifiers should not happen, as re-composing should be the approach taken... aaaaanyways haha the different structs approach is prefered for this case